target/linux/ramips/dts/mt7628an_hak5_wifi-pineapple-mk7.dts

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7628an.dtsi"
+
+/ {
+	compatible = "hak5,wifi-pineapple-mk7", "mediatek,mt7628an-soc";
+	model = "Hak5 WiFi Pineapple Mark 7";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x08000000>;
+	};
+
+	aliases {
+		led-boot = &led_status;
+		led-failsafe = &led_status;
+		led-running = &led_status;
+		led-upgrade = &led_status;
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		red {
+			label = "red:system";
+			gpios = <&gpio 0 0>;
+		};
+
+		green {
+			label = "green:system";
+			gpios = <&gpio 2 0>;
+		};
+
+		led_status: blue {
+			label = "blue:system";
+			gpios = <&gpio 3 0>;
+		};
+	};
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c", "p4led_an", "p0led_an", "p1led_an", "p2led_an", "p3led_an", "wdt", "refclk", "i2s";
+			ralink,function = "gpio";
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0xf80000>;
+			};
+		};
+	};
+};
+
+&ethernet {
+	status = "okay";
+	mtd-mac-address = <&factory 0x4>;
+};
+
+&wmac {
+	status = "okay";
+};
```

Comment:
What about the other Wifis?

Reply:
The MT7601's are connected via USB, and don't seem to require anything special in the DTS for them to be brought up by the mt76 driver. I will update the commit message to specify the MT7601**U** variant when I address the other requests.